### PR TITLE
feat(rti): pass JVM options to JAVA based federates

### PR DESCRIPTION
**Pull request by kschrab**
## Description

* This extends the JavaFederateExecutor to be able to pass JVM options.
* Furthermore, the RTI initialization now creates the `ComponentProvider` _before_ building the `FederateExecutors`.

## Issue(s) related to this PR

* Part of internal issue 861
  
## Affected parts of the online documentation

Changes in the documentation required?

No

## Definition of Done

**Prerequisites**

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] Your GitHub user id is linked with your Eclipse Account.

**Required**

- [x] The title of this merge request follows the scheme `type(scope): description`  (in the style of [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary))
- [ ] You have assigned a suitable label to this pull request (e.g., `enhancement`, or `bugfix`)
- [x] `origin/main` has been merged into your Fork.
- [x] Coding guidelines have been followed (see CONTRIBUTING.md).
- [ ] All checks on GitHub pass.
- [ ] All tests on [Jenkins](https://ci.eclipse.org/mosaic/job/mosaic/) pass.

**Requested** (can be enforced by maintainers)

- [ ] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [ ] If a bug has been fixed, a new unit test has been written (beforehand) to prove misbehavior
- [ ] There are no new SpotBugs warnings.

### CODE REVIEW

**Review by schwepmo — CHANGES_REQUESTED** [review]
LGTM! I left some minor remarks

File: rti/mosaic-rti-api/src/main/java/org/eclipse/mosaic/rti/api/federatestarter/JavaFederateExecutor.java [hunk 1]
```diff
@@ -125,38 +132,16 @@ public void stopRemoteFederate(PrintStream sshStreamOut) throws FederateStarterE
         }
     }
 
-    private String createClasspath(File workingDir, String fileSeparator, String pathSeparator) {
-        String jarName;
-        final StringBuilder classPath = new StringBuilder(".");
+    private String createClasspath(String fileSeparator, String pathSeparator) {
```
[the remark this answers is not in the document]
Comment (kschrab) [reply rti/mosaic-rti-api/src/main/java/org/eclipse/mosaic/rti/api/federatestarter/JavaFederateExecutor.java:135]: > Could there ever be an issue when using the relative paths to the root and lib directories?

What scenario do you think of? I assume that this feature is available in all runtime implementations. It was at least documented in the Java 8 docs: https://docs.oracle.com/javase/8/docs/technotes/tools/windows/classpath.html

File: rti/mosaic-rti-api/src/main/java/org/eclipse/mosaic/rti/api/federatestarter/JavaFederateExecutor.java [hunk 2]
```diff
@@ -101,11 +109,10 @@ public int startRemoteFederate(CLocalHost host, PrintStream sshStream, InputStre
             throw new FederateStarterException("Federate has been already started");
         }
 
-        final File workingDir = handle.getBinariesDir();
         final String sep = File.separator;
         final String fileSep = host.operatingSystem == CLocalHost.OperatingSystem.WINDOWS ? ";" : ":";
```
[the remark this answers is not in the document]
Comment (schwepmo) [reply rti/mosaic-rti-api/src/main/java/org/eclipse/mosaic/rti/api/federatestarter/JavaFederateExecutor.java]: Alright, but shouldn't the same logic apply to the file separator?

[the remark this answers is not in the document]
Comment (kschrab) [reply rti/mosaic-rti-api/src/main/java/org/eclipse/mosaic/rti/api/federatestarter/JavaFederateExecutor.java]: 🤔 good point

File: rti/mosaic-rti-api/src/main/java/org/eclipse/mosaic/rti/api/federatestarter/JavaFederateExecutor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (schwepmo) [reply rti/mosaic-rti-api/src/main/java/org/eclipse/mosaic/rti/api/federatestarter/JavaFederateExecutor.java:135]: Maybe I'm missing something but if MOSAIC is started through e.g., the SimulationRunner couldn't this path be wrong?

[the remark this answers is not in the document]
Comment (kschrab) [reply rti/mosaic-rti-api/src/main/java/org/eclipse/mosaic/rti/api/federatestarter/JavaFederateExecutor.java:135]: The paths specified here start at the execution directory of the federate, which is set in the `workingDir` variable. This working directory is set to the place where the federate.jar is located. This is either `tmp/federate-id` (only if `deploy=true`) or `bin/fed/federate-id`. The `lib` directory added to the classpath is _not_ the `lib` directory of MOSAIC, but (an optional) `lib` directory inside the federate directory (e.g. inside `bin/fed/federate-id`).

[the remark this answers is not in the document]
Comment (schwepmo) [reply rti/mosaic-rti-api/src/main/java/org/eclipse/mosaic/rti/api/federatestarter/JavaFederateExecutor.java:135]: Okay, thanks for clarifying. My worry was that with the removed `workingDir` parameter we would loose information. But this seems to have been unfounded!

**Review by schwepmo — APPROVED** [review]

**Pull request merged**